setuptools/command/build_ext.py

```diff
@@ -297,32 +299,33 @@ def _write_stub_file(self, stub_file: str, ext: Extension, compile=False):
         if not self.dry_run:
             f = open(stub_file, 'w')
             f.write(
-                '\n'.join([
-                    "def __bootstrap__():",
-                    "   global __bootstrap__, __file__, __loader__",
-                    "   import sys, os, pkg_resources, importlib.util" +
-                    if_dl(", dl"),
-                    "   __file__ = pkg_resources.resource_filename"
-                    "(__name__,%r)"
-                    % os.path.basename(ext._file_name),
-                    "   del __bootstrap__",
-                    "   if '__loader__' in globals():",
-                    "       del __loader__",
-                    if_dl("   old_flags = sys.getdlopenflags()"),
-                    "   old_dir = os.getcwd()",
-                    "   try:",
-                    "     os.chdir(os.path.dirname(__file__))",
-                    if_dl("     sys.setdlopenflags(dl.RTLD_NOW)"),
-                    "     spec = importlib.util.spec_from_file_location(",
-                    "                __name__, __file__)",
-                    "     mod = importlib.util.module_from_spec(spec)",
-                    "     spec.loader.exec_module(mod)",
-                    "   finally:",
-                    if_dl("     sys.setdlopenflags(old_flags)"),
-                    "     os.chdir(old_dir)",
-                    "__bootstrap__()",
-                    ""  # terminal \n
-                ])
+                '\n'.join(
```

Comment:
(Brainstorming)

In a future change, we probably want to rewrite this using f-strings to make it more readable, there are some line breaks that don't make much sense. It would also be nice if we can replace `pkg_resources` with `improtlib.resources`.

Reply:
I do agree with using f-strings. I've observed that `pyupgrade --py38-plus` automatically converts most formatting strings to f-strings, so maybe we should re-consider running that across the codebase.